nucliadb/src/nucliadb/writer/api/v1/knowledgebox.py

```diff
@@ -32,61 +38,101 @@
     KnowledgeBoxObjID,
     NucliaDBRoles,
 )
-from nucliadb_protos.knowledgebox_pb2 import (
-    DeleteKnowledgeBoxResponse,
-    KnowledgeBoxID,
-    KnowledgeBoxNew,
-    KnowledgeBoxResponseStatus,
-    KnowledgeBoxUpdate,
-    NewKnowledgeBoxResponse,
-    UpdateKnowledgeBoxResponse,
-)
+from nucliadb_protos import knowledgebox_pb2
 from nucliadb_utils.authentication import requires
-from nucliadb_utils.utilities import get_ingest
+from nucliadb_utils.settings import is_onprem_nucliadb
+
+
+def only_for_onprem(fun):
```

Comment:
NIT: decorators are cool but is it worth the complexity here? Maybe it's easier and more obvious to the reader if you add this in every endpoiint --  like you did for the grpc interface.
Feel free to disregard the comment though

Reply:
Sure, I can be explicit and add the conditionals in each function. I thought it was better to not duplicate the code, but if it's harder to read, I can change it